api/v1beta1/types.go

```diff
@@ -28,12 +28,19 @@ type OpenStackMachineTemplateResource struct {
 	Spec OpenStackMachineSpec `json:"spec"`
 }
 
+// ImageFilter describes the data needed to identify which image to use.
```

Comment:
One more thing: please can you summarise the validation requirements below in this user-visible godoc?